### PR TITLE
ESQL: Move `LogicalPlan` to esql proper

### PR DESCRIPTION
**Pull request by nik9000**
This moves `LogicalPlan` and all subclasses and references out of our esql-core and into esql-proper so that we can further simplify things.

### CODE REVIEW

**Comment by nik9000** [comment]
run elasticsearch-ci/part-2

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/OptimizerRules.java [hunk 1]
```diff
@@ -220,178 +180,6 @@ protected Expression maybeSimplifyNegatable(Expression e) {
...
-                    return skipPlan(limit);
-                }
-            }
-            return limit;
-        }
-
-        protected abstract LogicalPlan skipPlan(Limit limit);
-    }
-
     public static class FoldNull extends OptimizerExpressionRule<Expression> {
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/OptimizerRules.java:183]: To be consistent with the other optimization rules, maybe we should also extract the rules in this class into their own files and place them into `x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/`.

**Review by alex-spies — APPROVED** [review]
LGTM, thanks Nik!

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/analyzer/VerifierChecks.java [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-import org.elasticsearch.xpack.esql.core.plan.logical.LogicalPlan;
-
-import java.util.Set;
-
-import static org.elasticsearch.xpack.esql.core.common.Failure.fail;
-import static org.elasticsearch.xpack.esql.core.type.DataType.BOOLEAN;
-
-public final class VerifierChecks {
-
-    public static void checkFilterConditionType(LogicalPlan p, Set<Failure> localFailures) {
```
Comment (astefan) [inline x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/analyzer/VerifierChecks.java:22]: This one is in the Verifier itself in ESQL.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/OptimizerRules.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/OptimizerRules.java:183]: I agree, but I'm going to save that work for another time so I can get this in.

**Pull request merged**